### ISSUE TITLE
[windows] problem with py::arg("name")=value

### ISSUE THREAD

**Issue by looooo** [pybind/pybind11#469]
Recently I am trying to build some packages with windows. There I have encountered this problem:
```
4 overloads have similar conversions
c:\miniconda35\envs\_build\include\pybind11\attr.h(31): note: could be 'pybind11::arg &pybind11::arg::operator =(pybind11::arg &&)'
c:\miniconda35\envs\_build\include\pybind11\attr.h(31): note: or       'pybind11::arg &pybind11::arg::operator =(const pybind11::arg &)'
c:\miniconda35\envs\_build\include\pybind11\attr.h(26): note: or       'pybind11::arg_t<const char *> pybind11::arg::operator =<char,1>(const T (&)[1]) const'
```
build:
https://ci.appveyor.com/project/looooo/staged-recipes/build/1.0.28/job/2f0t1klmtruqbomq#L1084

source:
https://github.com/GrapeTec/T-SPLINE/blob/master/python/tspline_pyext.cpp#L202

It uses the conda-forge version of pybind11 (1.8-py35_0).

**Comment by dean0x7d** [comment]
This issue has been fixed in the master branch (#244), but it's not in a released version yet.
For a workaround, you should be able use `py::arg("name")=py::str("")`.

**Comment by looooo** [comment]
thanks, i will take the woraround.
